683  combined d10 upgrade

quicksketch opened the pull request:
Combines all the current Drupal 10 upgrade pull requests into a single big one:

- https://github.com/MukurtuCMS/Mukurtu-CMS/pull/698
- https://github.com/MukurtuCMS/Mukurtu-CMS/pull/700
- https://github.com/MukurtuCMS/Mukurtu-CMS/pull/701
- https://github.com/MukurtuCMS/Mukurtu-CMS/pull/699

On my local, this actually correctly installs Drupal 10.4.1 and seems work.

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
Running issues list. To be expanded and detailed in dedicated issues.

- [ ] UID 1 warning message styling in Mukurtu theme is wrong (fine in admin theme).
- [ ] Homepage styling isn't complete anyway, but the blocks are repeating.
- [ ] User menu link PR needs to be replicated: https://github.com/MukurtuCMS/Mukurtu-CMS/pull/646
- [ ] Browse/search pages (Search API, not Solr) are not returning any results.
- [ ] System is trying to auto-assign protocols when uploading multiple media assets in one batch.
- [ ] Media library not showing any previews.
- [ ] No PDF preview thumbnail.

michael-wynne-wsu commented:
@quicksketch I'll try to poke at some more things later, but honestly not much has jumped out at me beyond what we already discussed. There was already gaps due to lack of theme and feature completeness, but very few "new" bugs since the last stable 9.5 build.

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
@quicksketch I haven't identified any other functional issues introduced in this PR except for search/browse not returning any results. We have no issues with merging the PR and resolving the search issue in D10 environment, but if you want to address it first, we'll defer to you!

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

quicksketch commented:
@michael-wynne-wsu I fixed most of the issue I believe. Could you validate?

> UID 1 warning message styling in Mukurtu theme is wrong (fine in admin theme).

This seems to be because messages are loaded after the initial page load in D10. I disabled the "Big Pipe" module that does this lazy-loading of page parts. The Big Pipe module is not compatible with some hosting (like Pantheon) and can affect page caching. We turn it off on all our sites at Lullabot as far as I know.

> Homepage styling isn't complete anyway, but the blocks are repeating.

Fixed these duplicate blocks. Somehow the config was duplicated for placing blocks. I removed the duplicate files.

> Browse/search pages (Search API, not Solr) are not returning any results. /browse, /dictionary, /digital-heritage, /collections

I think these are all fixed now. I installed the site as Drupal 9, switched the branch, then ran update.php to capture config changes. All config changes are now re-exported after the upgrade. That seemed to fix most listings.

The `/dictionary` listing seemed to have a broken configuration for the "Word Type" facet. I fixed this configuration manually based on the way that the "Word List" filter was set up. I believe it's working as intended now.



> User menu link PR needs to be replicated:
> https://github.com/MukurtuCMS/Mukurtu-CMS/pull/646
> System is trying to auto-assign protocols when uploading multiple media assets in one batch.
> Media library not showing any previews.
> No PDF preview thumbnail.

I have not addressed these items yet, but some(?) might be fixed by the updated configuration being exported.

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-1vclypic3ftxnnor8bikyyb82rxtvpdc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67a40238b029347f4b8438f4

alexmerrill commented:
@quicksketch  I see the UID message as fixed as well as the duplicate blocks on home page -- I am still not able to browse or search for items  @ /digital-heritage or /browse

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-9dmnpw1idn24hgka0l9cas7blp5rmlcv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67b8c6a8b029347f4beddb5e

quicksketch commented:
Thank you for testing @alexmerrill! I should have looked at the Tugboat sandbox after pushing my changes. The Tugboat sandbox needed to be rebuilt to show all the changes. I rebuilt it and now the search pages should be working. Can you check again? Thanks!

quicksketch commented:
>  Media library not showing any previews.

This looks like it's because our Tugboat installations aren't supporting WebP, which Drupal core now uses by default as of Drupal 10.3: https://www.drupal.org/project/drupal/issues/3406267. I'll fix that in a separate issue.

alexmerrill commented:
> Thank you for testing @alexmerrill! I should have looked at the Tugboat sandbox after pushing my changes. The Tugboat sandbox needed to be rebuilt to show all the changes. I rebuilt it and now the search pages should be working. Can you check again? Thanks!

Yup -- using tugboat seeing the updates/fixes thanks!

quicksketch commented:
> User menu link PR needs to be replicated

Elaborating on this, the "Add Content" menu link is missing. It should look like this:

![image](https://github.com/user-attachments/assets/a90d2e9c-1d89-4d2c-a315-eec882cd3281)

But the underlying issue here is the `/dashboard/node/add` path is giving a 403 access denied. That's why the link is hidden. This is because our custom menu callbacks are using [a deprecated access check](https://www.drupal.org/node/2836069). I'll push a commit to fix those.

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr702-9dmnpw1idn24hgka0l9cas7blp5rmlcv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67b8c6a8b029347f4beddb5e

quicksketch commented:
I think all known issues are now addressed.

> System is trying to auto-assign protocols when uploading multiple media assets in one batch.

I tried to reproduce this problem but it _might_ be fixed now?

- Visit https://mukurtu4.ddev.site/dashboard/node/add/person
- Click "Add Media" within "Media Assets"
- Click the "Image" tab
- Click "Browse" within the file upload. Select 5 different images at once.
- Upload, then confirm that the media form is shown for all 5 images.
- Select a different protocol (or no protocol at all) for the images.
- Verify that validation fires indicating protocol must be set per image. Validate different protocols can be set per image.

This all seems to work, but I did not specifically fix this issue. It may have been fixed as a side-effect.

michael-wynne-wsu commented:
@quicksketch I think that issue IS resolved. Apologies for the incomplete notes, I had meant to add more before I left. When uploading media assets of multiple types, the selected protocol was carrying over, but I'm not seeing it now at all.

I don't see any remaining open issues with this PR!
